src/charts/stackedArea/stackedAreaComponent.js

```diff
@@ -1 +1,24 @@
-console.log('hello stacked area');
+import React from 'react';
+import stackedArea from './stackedAreaChart';
+
+
+export default class StackedArea extends React.Component {
+
+    componentDidMount() {
+        stackedArea.create(this._rootNode, {
+            data: this.props.data,
+        });
+    }
+
+    _setRef(componentNode) {
+        if (componentNode) {
+            this._rootNode = componentNode;
+        }
+    }
```

Comment:
Should we eventually extract some of the logic from `stackedAreaComponent` and `stackedAreaChart` into something more generic that can be used with other chart types?

I think the other charts are going to have a lot in common, for example the `stackedArea.create` function will probably look very similar for bar chart and line chart, probably others too? `stackedAreaComponent`'s `render, _setRef, and componentDidMount` is also going to be pretty much the same for other charts.

Could do it either in a OOP way or HOC way. `baseComponent` as a parent class for `stackedAreaComponent`, or provide the `baseComponent` logic as a HOC.

I'm guessing the update logic is going to be what varies the most between chart types, so the children classes would mainly be focused on handling that.

Reply:
Yeah, we will probably do something about it in the future, but first, we need to see the whole extent of our components and compare them, so we can extract the right abstraction from them.